frontend/src/scenes/data-warehouse/editor/QueryWindow.tsx

```diff
@@ -0,0 +1,175 @@
+                codeEditorProps={{
+                    onChange: (v) => {
+                        setQueryInput(v ?? '')
+                        updateState()
+                    },
+                    onMount: (editor, monaco) => {
+                        setMonacoAndEditor([monaco, editor])
+
+                        const allModelQueries = localStorage.getItem(editorModelsStateKey(codeEditorKey))
+                        const activeModelUri = localStorage.getItem(activemodelStateKey(codeEditorKey))
+
+                        if (allModelQueries) {
+                            // clear existing models
+                            monaco.editor.getModels().forEach((model) => {
+                                model.dispose()
+                            })
+
+                            const models = JSON.parse(allModelQueries || '[]')
+                            const newModels: Uri[] = []
+
+                            models.forEach((model: Record<string, any>) => {
+                                if (monaco) {
+                                    const uri = monaco.Uri.parse(model.path)
+                                    const newModel = monaco.editor.createModel(model.query, 'hogQL', uri)
+                                    editor?.setModel(newModel)
+                                    newModels.push(uri)
+                                }
+                            })
+
+                            setModels(newModels)
+
+                            if (activeModelUri) {
+                                const uri = monaco.Uri.parse(activeModelUri)
+                                const activeModel = monaco.editor
+                                    .getModels()
+                                    .find((model) => model.uri.path === uri.path)
+                                activeModel && editor?.setModel(activeModel)
+                                const val = activeModel?.getValue()
+                                if (val) {
+                                    setQueryInput(val)
+                                    saveQuery()
+                                }
+                                setModel(uri)
+                            } else if (newModels.length) {
+                                setModel(newModels[0])
+                            }
+                        } else {
+                            const model = editor.getModel()
+
+                            if (model) {
+                                addModel(model.uri)
+                                setModel(model.uri)
+                            }
+                        }
+                    },
+                    onPressCmdEnter: (value, selectionType) => {
+                        if (value && selectionType === 'selection') {
+                            saveQuery(value)
+                        } else {
+                            saveQuery()
+                        }
+                    },
```

Comment:
I know this is the existing logic right now, but I wanna get away from this model of being able to "save" a selection of the query from the editor - this sets the inner `setQuery` of `hogQLQueryEditorLogic`. 

Ultimately, I'd like a pattern here where the HogQL query is separated from the editor itself. We have a bunch of workarounds and less-than-ideal UX due to how the editor logic works in conjunction with `dataNodeLogic`, and this is a chance to completely rewrite that. An example of this is: if you update the text in the hogql editor and don't run it but save it - the query text of the last run is whats saved and not whats currently in your editor = bad and weird UX. 

I'd consider thinking about writing a new logic for managing all of this model state along with the appropriate components - the current set up really isn't designed with tabs in mind. We can fix this now before its too late and becomes a tough situation later

Reply:
> if you update the text in the hogql editor and don't run it but save it - the query text of the last run is whats saved and not whats currently in your editor = bad and weird UX.

This actually isn't an issue with how I used datanodelogic right now. DataNodeLogic right now is mainly used to just be fed a query and return some results. The actual query that's handled by the saving vs the editing is decoupled with a component state var `setActiveQueryInput`. 

But, overall I do agree this logic needs some rethinking. I worked this in for now to get an idea of dependencies of the current editor but definitely open to making a new logic as there's a good bit of logic that's sitting inside the component/onMount handler